packages/core/src/chain.ts

```diff
@@ -36,7 +36,7 @@ async function setChain(options: {
 
   // validate that chainId has been added to chains
   const chain = chains.find(
-    ({ namespace, id }) => namespace === chainNamespace && id === chainIdHex
+    ({ namespace, id }) => namespace === chainNamespace && id.toLowerCase() === chainIdHex.toLowerCase()
```

Comment:
```suggestion
    ({ namespace, id }) =>
      namespace === chainNamespace &&
      id.toLowerCase() === chainIdHex.toLowerCase()
```
A small update to match our formatting